### DISCUSSION TITLE
Explorando Inovações para a Versão 2 (V2) do Diciotech 🚀✨

### DISCUSSION THREAD

**Discussion by levxyca** [levxyca/diciotech#44] [Ideas]
**Construindo juntes a versão 2 (V2) do projeto: compartilhe suas ideias e feedbacks!**

A v1 tem sido um sucesso graças a todas as pessoas contribuidoras! Agora, é hora de olharmos adiante e discutirmos o que podemos fazer ainda melhor na segunda versão. Queremos ouvir suas sugestões não apenas para novos recursos, mas também para aprimoramentos na organização e escalabilidade do projeto.

Até agora, na v1, já conquistamos (tanto em coisas que estão prontas, quanto coisas em andamento):

- Filtro
- Favoritar cards
- Modo escuro
- GitHub Corner
- Rolagem suave nos cards

Além das melhorias visuais e de usabilidade, também adicionamos novos termos.

Estamos animades com o progresso da nova UI #12 e agora, queremos abrir esse espaço para todas as pessoas. Compartilhe suas ideias, feedbacks e tudo mais que estiver no seu coração. Juntes, vamos fazer da V2 algo incrível! 💡👏"

**Comment by levxyca** [comment]
Alguns itens que eu gostaria de implementar no projeto, pensando em organização e melhor transparência:
- Template de issue
- Template de PR
- Sempre criar issue antes de um PR
- Criar um wiki para o projeto
- Manter o ROADMAP vivo e atualizado
- Adicionar um código de conduta
- Adicionar uma licença

**Reply by freitaschz** [reply]
Essas implementações seriam muito importantes e trariam uma organização muito maior ao projeto! Eu tinha até feito um template de issue e PR para o projeto, mas acabei nem abrindo uma issue própria sobre o assunto.

Para complementar, também creio que é importante impor algumas regras e requisitos para PRs de implementação de novos termos no Diciotech, conforme sugeri nas replies na issue #54.

**Reply by freitaschz** [reply]
@levxyca, com relação ao template de PR e de issue, além do código de conduta, todos irão ser feitos e adicionados por ti ou vais querer faze-los junto a comunidade?

**Reply by levxyca** [reply]
@thiagofqs sobre os templates, a ideia é faze-los junto com a comunidade sim :) acabei de atualizar o roadmap do projeto e a ideia é ir criando novas issues para ir implementando isso durante a v2.

**Comment by Ryrden** [comment]
Posso cuidar da licença? Qual tipo de permissividade você quer para esse projeto? eu acho que uma creative commons, GPL ou apache seja uma boa :)

**Reply by levxyca** [reply]
Claro! Vou ver se consigo no final dessa semana, ou mais tardar no inicio da semana que vem, mexer no ROADMAP do projeto e começar a criar as issues, aí já conseguimos começar a andar com a release 2.

Sobre o tipo de licença, para ser sincera, preciso dar uma lida sobre as permissividades para entender melhor!

**Reply by freitaschz** [reply]
O próprio GitHub, em sua documentação sobre [Licenciar um repositório](https://docs.github.com/pt/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/licensing-a-repository), sugere o [choosealicense.com](https://choosealicense.com/) para obter as licenças disponíveis e também para ajudar a entender como licenciar um código. Dá uma conferida lá depois!

**Reply by Ryrden** [reply]
Esse site é ótimo, iai @levxyca, alguma decisão?

**Reply by levxyca** [reply]
Sim, acabei de criei um issue para isso #65! Se quiserem conversar mais sobre esse ponto da licença, podemos levar essa discussão para a issue.

**Comment by levxyca** [comment]
Passando para avisar que estou atualizando e organizando o projeto, tendo isso em vista, atualizei o nosso [ROADMAP](https://github.com/levxyca/diciotech/blob/main/ROADMAP.md), se quiserem dar um bizu :)

**Comment by george-gca** [comment]
Com relação aos templates de issues, eu acho interessante a solução que usam no [pytorch-lightning](https://github.com/Lightning-AI/pytorch-lightning/issues/new/choose). Eles têm [vários templates](https://github.com/Lightning-AI/pytorch-lightning/tree/master/.github/ISSUE_TEMPLATE) dependendo do tipo de issue, e acho que alguns seriam bem úteis pra gente que não mexe muito com código conseguir contribuir. O template é em markdown assim, por exemplo:

![image](https://github.com/levxyca/diciotech/assets/31376482/1102876a-b377-4504-a026-ad5c6cb9a51e)

**Reply by levxyca** [reply]
Sensacional! Vou dar uma melhor olhada em como eles fazem e pegar algumas inspirações!

**Reply by george-gca** [reply]
Sobre templates de PRs, tem uma menor e mais autocontida interessante que é a do [pytorch lightning](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/PULL_REQUEST_TEMPLATE.md) e uma mais completa como a do [Forem](https://github.com/forem/forem/blob/main/.github/PULL_REQUEST_TEMPLATE.md).

**Comment by andrepg** [comment]
Hey @levxyca, como anda? 

Estava pesquisando algumas ideias e encontrei este repositório. Vi que tem muitos PRs abertos e o redesenho programado. Também queria começar a sugerir algumas melhorias via discussão. Como estamos nesse sentido? Há algum tempo não temos PRs fechados e queria me atualizar...

**Comment by levxyca** [comment]
Hey @andrepg, tudo bem?

O projeto realmente está com muitos PR's abertos e atuamos conforme conseguimos 😅 Fique muito a vontade para começar a sugerir algumas melhorias via discussão! Também sinta-se muito livre se tiver sugestões com os PR's que temos abertos ou até mesmo com ajuda para revisão 🫰🏻🚀

**Reply by andrepg** [reply]
Até tenho um par de sugestões para fazer/planos para ação. No entanto, acredito que seria legal verificar o andamento da v2 ou fecharmos os PRs de adição de termos para começar a reestruturar o dicionário. 

Eu não gostaria, por exemplo, de simplesmente modificar o PR já atribuído a outro colega, mas alguns há tempos não tem atualizações. Mantenhamos contato, eu tenho um projeto semelhante no meu Notion, mas o seu viu a luz do dia antes. Então, seguimos por aqui melhorando a comunidade! 

Por outro lado, acredito que posso começar com os modelos de issue para novos termos e de bugs. São bem simples, na verdade.

E se não deixei claro o suficiente, se precisar de mim em algo específico, é só dar um “grito” hehe… 

Um abraço,

**Reply by george-gca** [reply]
Acredito que a maioria das PRs (assim como a minha #150) são adições de definições, então é algo mais fácil de resolver.

**Reply by andrepg** [reply]
@george-gca este nem é um problema tão grande. Aprovar e fechar PRs de definição é simples. O problema começa quando há correções solicitadas, como o #119, o #125 e outros. Correções essas que são basicamente conflitos de merge porque temos tudo junto 😆 

Tiro um tempo para olhar os mais antigos e fazer uma limpa em breve.

**Reply by george-gca** [reply]
É, como todas as definições estão no mesmo arquivo, um merge manual é quase inevitável. Pior que nem sei dizer se vale a pena começar a quebrar em vários arquivos ou usar alguma outra solução nesse momento.

**Reply by levxyca** [reply]
@andrepg @george-gca bacana demais a conversa e sugestões de vocês pessoal! 🫰🏻estive pensando aqui, acho que o primeiro passo seria fazer uma limpa nessas PR's né? Não qual a melhor opção, fechar as PR's que estão a muito tempo sem atualizações e correções, talvez? Acho que as que não foram mergeadas apenas por conflitos, eu poderia eu mesma atualizar eles, pois não iria interferir no que foi feito pela pessoa contribuidora, agora os que tinham correções solicitadas acho que talvez o melhor é só fechar a PR e explicar, o que acham?

@andrepg enquanto isso, fique a vontade para começar com os modelos de issue para novos termos e de bugs, por favor!

a partir daí, vale a gente começar a olhar o andamento da v2, atualizar o roadmap também, tá bem desatualizado 🥲 e começar a entender essas questão de organização de arquivos e como podemos melhorar 🤔

**Reply by george-gca** [reply]
Acho que se a correção for algo pequeno (e a branch que a pessoa abriu permitir edição pelos mantenedores) acho que vale fazer a correção e aplicar. Ou então comentar na PR marcando quem criou e pedindo pra fazer updates. Nesse ponto acho que vale implementar algo como um `stale` pras PRs. Acabei de ver que [o próprio GitHub tem um workflow](https://github.com/actions/stale) pra isso.

**Reply by andrepg** [reply]
Concordo com o @george-gca. São muitas reviews solicitados que nunca foram alterados, principalmente. Outras poucas coisas do visual e a tão esperada v2. 

Depois disso podemos abrir uma discussão para verificar a melhor estrutura de dados.

**Comment by levxyca** [comment]
@george-gca @andrepg bacana pessoal, vou atuar em cima disso 🚀

**Comment by AdrianaQMelo** [comment]
@levxyca uma ideia que se passou aqui... Que tal uma gameficação do tipo cards de memorização, não lembro o nome da técnica, mas é uma que na frente aparece o termo, você responde e depois vira o card com a "Resposta correta". Creio que hoje é um jeito muito legal de aprender, usei muito isso quando estava estudando idiomas 😊

**Comment by levxyca** [comment]
@AdrianaQMelo essa ideia é muito legal! eu já usei alguns apps que faziam algo parecido 👀 gostei!

**Comment by george-gca** [comment]
Eu acho que seria legal se isso fosse um modo ativavel e não o padrão. Tipo um modo "me ajude a lembrar os termos", algo assim.

**Reply by AdrianaQMelo** [reply]
@george-gca Concordo, até porque pessoas estudam de maneiras diferentes, podemos entrar talvez na ideia de: "Game mode", já que o projeto tem o Light e dark mode ksks

**Reply by levxyca** [reply]
Sim sim, tb acho super de acordo. Uma coisa que eu queria tb, falando em diferentes formas de estudos, era conseguirmos adicionar diferentes tipos de conteúdos nos cards sabe? Linkar uma documentação, a pessoa poder assistir um vídeo, ver uma imagem ilustrativa, algo de acordo com a preferência, sabe?

**Reply by AdrianaQMelo** [reply]
Amei a ideia <3 você mesma anda coordenando para abrir as issues ou tem uma equipe de discovery contigo @levxyca ?

**Reply by levxyca** [reply]
@AdrianaQMelo acaba sendo eu mesmo, mas sempre tem alguma pessoa me ajudando como o @george-gca ue está sempre por aqui 😊 inclusive sempre estou aberta a novos voluntários 😝
